sdk/js/src/EventQueue.ts

```diff
@@ -74,31 +101,40 @@ export class EventQueue {
         this.eventQueue = []
         this.aggregateEventMap = {}
 
-        try {
-            const res = await publishEvents(
-                this.sdkKey,
-                this.client.config || null,
-                user,
-                eventsToFlush,
-                this.client.logger,
-            )
-            if (res.status !== 201) {
-                this.eventQueue.push(...eventsToFlush)
-            } else {
-                this.client.logger.info(
-                    `DVC Flushed ${eventsToFlush.length} Events.`,
+        const eventRequests = chunk(eventsToFlush, this.eventQueueBatchSize)
+        for (const eventRequest of eventRequests) {
+            try {
+                const res = await publishEvents(
+                    this.sdkKey,
+                    this.client.config || null,
+                    user,
+                    eventRequest,
+                    this.client.logger,
                 )
+                if (res.status !== 201) {
+                    this.eventQueue.push(...eventRequest)
+                } else {
+                    this.client.logger.info(
+                        `DVC Flushed ${eventRequest.length} Events.`,
+                    )
+                }
+            } catch (ex) {
+                this.client.eventEmitter.emitError(ex)
+                this.eventQueue.push(...eventRequest)
```

Comment:
so if the code fails to publish the events, the data goes back into the queue for a retry on a future flushEvents() call and we continue processing the other chunks. Any issues with events getting out of order or are they all fully timestamped so that they will resync in snowflake?

Also is there an potential for bad data to continually cause those events to fail and just keep getting requeued forever?

Reply:
yea they are time-stamped when they are created. any yea failed events are a possible problem. let me investigate that.

Reply:
to be fair, this might be an issue with all of our SDKs that use a similar pattern. i think we generally report failed payloads back to WASM and it just re-queues them

Reply:
in the Node SDK we have a whole complicated payload retry system. However for the JS SDK I think I'm just going to say we drop `4XX` errors and retry `5XX` errors.